fedora_messaging/cli.py

```diff
@@ -195,3 +196,99 @@ def publish(exchange, file):
             except exceptions.ConnectionException as e:
                 click.echo("Unable to connect to the queue: {}".format(str(e.reason)))
                 sys.exit(errno.ECONNREFUSED)
+
+
+@cli.command()
+@click.argument("file", type=click.File("w"))
+@click.option("--limit", help=_limit_help, type=click.IntRange(1))
+@click.option("--app-name", help=_app_name_help, default="recorder")
+@click.option("--routing-key", help=_routing_key_help, multiple=True)
+@click.option("--queue-name", help=_queue_name_help)
+@click.option("--exchange", help=_exchange_help)
+def record(exchange, queue_name, routing_key, app_name, limit, file):
+    """Record messages from an AMQP queue to provided file."""
+
+    class Recorder:
```

Comment:
So what I'd recommend here is pulling the class out of the function. It's not *wrong*, but I think it's a little unexpected (I usually only define classes and functions in a parent class/function scope it if is both tiny and relevant only to the internal workings of the parent). Defining this class at the module level should be fine.

After that, I think you can get rid of pretty much all the rest of the code in the record command and have it just call the "consume" cli command here. The only tricky bit is the consume call expects the callback to be a path. I think you could either hard-code the path or add a check to the consume code to see if it's already a class/function. The only downside to hard-coding I see is if the module name changes it'll break, but that will also break APIs so I don't think that's a huge concern.

Reply:
But the question is how to pass limit and file to Record instance(instance of class outside record func). As I remember consumer api is responsible to make instance of class . Mabye classmethod?

Reply:
Ah, right. Maybe cli.consume() should accept a class/callable then, since you can create an instance of the class with the limit and file and pass it on.

Reply:
Also a good point. Given those things maybe leaving it as-is is fine. Or, since testing it here is a bit troublesome, perhaps pull it out into a function called ``recorder_factory`` or something which accepts the limit as an argument and returns the class with the limit baked in:

```
def recorder_factory(limit):
    class Recorder:
        def __init__(self):
             self.limit = limit
    return Recorder
```
Perhaps a future enhancement to api.consume would be to allow passing instances of classes. I'd say we shouldn't deal with that right now.